spec/features/user/articles/remove_spec.rb

```diff
@@ -1,40 +1,24 @@
 require "rails_helper"
 
 feature "Remove Own Article" do
-  include_context "current user signed in"
-  let!(:another_user) { create :user }
-  let!(:article) { create :article, user: current_user }
-  let!(:another_article) { create :article, user: another_user }
+  include_context "Full context data"
   let(:decorated_article) { article.decorate }
 
   background do
+    login_as current_user
     visit articles_path
   end
 
   subject { page }
 
-  context "links" do
-    it "exist by own article" do
-      within "#article_#{article.id}" do
-        is_expected.to have_content I18n.t "app.actions.remove"
-      end
-    end
-
-    it "doesn't exist by another article" do
-      within "#article_#{another_article.id}" do
-        is_expected.not_to have_content I18n.t "app.actions.remove"
-      end
-    end
-  end
-
-  context "result after action" do
+  context "remove action" do
```

Comment:
`context` should start with when or with. Please look here -> http://betterspecs.org/ru/